### DISCUSSION TITLE
XT emulation issues on Mac

### DISCUSSION THREAD

**Discussion by tyszler2** [86Box/86Box#3003] [Q&A] [answered]
Dear Group,

I am having a lot of issues with my 8088 4.77MHz XT emulation on Mac OS. I noticed that if I connected a hard disk it would list all of the contents accurately with DIR in DOS (3.10). After a short period of time it would list as nonsense. A reboot clears and the problem comes back. This occurred with MFM and IDE. I got an original diagnostics image (v2) and I get error after error. Some have to do with "timing" and "wait state". The configuration is as bare bones as possible. Still errors. 

The issue doesn't seem to affect floppy.

Should I expect that the diagnostics will pass? What could I do to get this working?

Thanks!

**Comment by TC1995** [comment]
chs of the hdd?

**Reply by tyszler2** [reply]
306/4/26 - however the problem still exists with the hdd controller and disk removed from the config. Thanks!

**Reply by tyszler2** [reply]
![Screen Shot 2022-12-29 at 2 18 39 PM](https://user-images.githubusercontent.com/121637799/210000536-458e6356-2319-49d9-9c0a-7697bf9b2faf.jpg)

**Reply by tyszler2** [reply]
and so on

**Reply by tyszler2** [reply]
The detected configuration

![Screen Shot 2022-12-29 at 2 17 21 PM](https://user-images.githubusercontent.com/121637799/210001253-f208ef5b-3dff-4c6e-b2e1-017fddc48b38.jpg)

**Comment by TC1995** [comment]
Don't bother with the diagnostics on the XT.

**Reply by tyszler2** [reply]
Thanks, I was emboldened by that demo on youtube as to the fidelity of the emulation. So why does my hard disk get scrambled? Pretty repeatable. I'll attach a screenshot when I get dos reinstalled on it.

**Reply by tyszler2** [reply]
btw I was getting an adapter error 1701 on POST using the MFM adapter so I switched to IDE, which did boot. I am struck by a little nostalgia for the PC/XT I grew up with. I am trying to duplicate its configuration. It was MFM for sure, had a sixpack, CGA, etc. was pre-1986.

**Reply by TC1995** [reply]
where have you put the MFM disk using the mfm controller and which controller? I mean, a mfm hdd should be on 0:0.

**Reply by tyszler2** [reply]
I figured out the adapter error. my old disk was 10mb, I had it at 15. maybe that is the whole issue?

**Reply by tyszler2** [reply]
I was using the IBM original MFM, I just saw that it only supports 10MB in another guide. My memory was faulty lol.

**Reply by TC1995** [reply]
it also supports 20MB (615/4/17).

**Reply by tyszler2** [reply, marked as answer]
@TC1995 The FDD controller was the issue all along. wrong geometry for the IBM MFM, then the pc/xt ide I tried caused the scrambling after a period of time. changed to 10MB disk and it is running just fine. 

Thanks so much for putting me on the right track!!!

**Reply by TC1995** [reply]
no problem.

**Reply by tyszler2** [reply]
I tried the 20MB (615/4/17) config and it throws the same error. What adapter would work reliably to go higher?

**Reply by TC1995** [reply]
20MB works okay with the IBM Xebec one on my end.

**Reply by tyszler2** [reply]
NM it seems the option on the drive got set to IDE inadvertently
